apollo-biz/src/test/java/com/ctrip/framework/apollo/biz/service/NamespaceBranchServiceTest.java

```diff
@@ -115,10 +122,24 @@ public void testUpdateBranchGrayRulesWithUpdateTwice() {
     Assert.assertEquals(2, releaseHistories.getTotalElements());
     Assert.assertEquals(ReleaseOperation.APPLY_GRAY_RULES, firstReleaseHistory.getOperation());
     Assert.assertEquals(ReleaseOperation.APPLY_GRAY_RULES, secondReleaseHistory.getOperation());
-    Assert.assertTrue(firstReleaseHistory.getOperationContext().contains(firstRule.getRules()));
-    Assert.assertFalse(firstReleaseHistory.getOperationContext().contains(secondRule.getRules()));
-    Assert.assertTrue(secondReleaseHistory.getOperationContext().contains(firstRule.getRules()));
-    Assert.assertTrue(secondReleaseHistory.getOperationContext().contains(secondRule.getRules()));
+    Assert.assertTrue(containRule(firstReleaseHistory.getOperationContext(), firstRule.getRules()));
+    Assert.assertFalse(containRule(firstReleaseHistory.getOperationContext(), secondRule.getRules()));
+    Assert.assertTrue(containRule(secondReleaseHistory.getOperationContext(), firstRule.getRules()));
+    Assert.assertTrue(containRule(secondReleaseHistory.getOperationContext(), secondRule.getRules()));
+  }
+
+  private boolean containRule(String context, String rule) {
```

Comment:
How about we refactor this method a little bit to actually check whether the context contains some rule? It's not necessary for current test cases, but we could use that in the future.

```java
  private boolean containRules(String context, String rules) {
    Type grayReleaseRuleItemsType = new TypeToken<Map<String, Set<GrayReleaseRuleItemDTO>>>() {
    }.getType();
    Map<String, Set<GrayReleaseRuleItemDTO>> contextRulesMap = new Gson().fromJson(context, grayReleaseRuleItemsType);
    Set<GrayReleaseRuleItemDTO> ruleSet = GrayReleaseRuleItemTransformer.batchTransformFromJSON(rules);

    for (GrayReleaseRuleItemDTO rule : ruleSet) {
      boolean found = false;
      for (Set<GrayReleaseRuleItemDTO> contextRules : contextRulesMap.values()) {
        for (GrayReleaseRuleItemDTO contextRule : contextRules) {
          if (contextRule.toString().equals(rule.toString())) {
            found = true;
            break;
          }
        }
      }
      if (!found) {
        return false;
      }
    }

    return true;
  }
```

Reply:
Yes, it looks excellent! This version does make more sense and makes the tests more consistent. Also thanks for fixing some style issues here :)